### ISSUE TITLE
Integrating with Gradle sample code no longer works with the Android plugin 7.3.1

### ISSUE THREAD

**Issue by Lonami** [mozilla/uniffi-rs#1386]
I believe it is [a bug in the Android Gradle Plugin 7.3.1](https://issuetracker.google.com/issues/255784699), but technically speaking, if it's not, then this project's documentation should be updated to reflect the new way in which integration with Gradle should be made. 
<sub>(My main motivation to post this is to save the evening of any other poor soul who encounters the same issue I did.)</sub>

The [Integrating with Gradle](https://mozilla.github.io/uniffi-rs/kotlin/gradle.html) page of the manual ([docs/manual/src/kotlin/gradle.md](https://github.com/mozilla/uniffi-rs/blob/main/docs/manual/src/kotlin/gradle.md)) has the following code:

```gradle
android.libraryVariants.all { variant ->
    def t = tasks.register("generate${variant.name.capitalize()}UniFFIBindings", Exec) {
        workingDir "${project.projectDir}"
        // Runs the bindings generation, note that you must have uniffi-bindgen installed and in your PATH environment variable
        commandLine 'uniffi-bindgen', 'generate', '<PATH TO .udl FILE>', '--language', 'kotlin', '--out-dir', "${buildDir}/generated/source/uniffi/${variant.name}/java"
    }
    variant.javaCompileProvider.get().dependsOn(t)
    def sourceSet = variant.sourceSets.find { it.name == variant.name }
    sourceSet.java.srcDir new File(buildDir, "generated/source/uniffi/${variant.name}/java")
    // XXX: I've been trying to make this work but I can't, so the compiled bindings will show as "regular sources" in Android Studio.
    idea.module.generatedSourceDirs += file("${buildDir}/generated/source/uniffi/${variant.name}/java/uniffi")
}
```

(In my case I had to replace `libraryVariants` with `applicationVariants` but that's just the way my project is structured.)

Obviating the `idea.module` line which I commented out, the line:

```gradle
sourceSet.java.srcDir new File(buildDir, "generated/source/uniffi/${variant.name}/java")
```

fails with the rather obscure error:

```
A problem occurred configuring project ':app'.
> Cannot convert the provided notation to a File or URI: [C:\...\app\build\generated\source\uniffi\debug\java].
  The following types/formats are supported:
    - A String or CharSequence path, for example 'src/main/java' or '/usr/include'.
    - A String or CharSequence URI, for example 'file:/usr/include'.
    - A File instance.
    - A Path instance.
    - A Directory instance.
    - A RegularFile instance.
    - A URI or URL instance.
    - A TextResource instance.

* Try:
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Exception is:
org.gradle.api.ProjectConfigurationException: A problem occurred configuring project ':app'.
```

I have tried many things but was unable to work around it, despite [the documentation for 7.3 `AndroidSourceDirectorySet`](https://developer.android.com/reference/tools/gradle-api/7.3/com/android/build/api/dsl/AndroidSourceDirectorySet) to still have the `srcDir`, so it appears the code is correct, but the plugin is buggy.

I have temporarily worked around this by downgrading the plugin to 7.2.2.

I would probably wait it out and see if this bug is resolved on their end, but another option would be to add a note in the manual saying "beware this code was tested with version X and may not work in newer versions".

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/UNIFFI-207)
┆friendlyId: UNIFFI-207

**Comment by stan-irl** [comment]
also seeing this

**Comment by Lonami** [comment]
This is apparently fixed in https://android.googlesource.com/platform/tools/base/+/7a6627a92857e772c378ddcf49546adf4ade53b5 available in versions 7.4 and 8.0 so this issue can be closed.
